app/bootstrap/cmd/main.go

```diff
@@ -16,28 +16,32 @@ import (
 	"github.com/spiffe/spike/internal/config"
 
 	"github.com/spiffe/spike/app/bootstrap/internal/env"
+	"github.com/spiffe/spike/app/bootstrap/internal/k8s"
 	"github.com/spiffe/spike/app/bootstrap/internal/net"
 	"github.com/spiffe/spike/app/bootstrap/internal/state"
 	"github.com/spiffe/spike/app/bootstrap/internal/url"
 )
 
 func main() {
-	const fName = "boostrap.main"
+	const fName = "bootstrap.main"
 
-	log.Log().Info(fName, "message", "Starting SPIKE bootstrap...", "version", config.BootstrapVersion)
+	log.Log().Info(fName, "message",
+		"Starting SPIKE bootstrap...",
+		"version", config.BootstrapVersion,
+	)
 
 	init := flag.Bool("init", false, "Initialize the bootstrap module")
 	flag.Parse()
 	if !*init {
 		fmt.Println("")
 		fmt.Println("Usage: bootstrap -init")
 		fmt.Println("")
-		os.Exit(1)
+		log.FatalLn(fName, "message", "Invalid command line arguments")
 		return
 	}
 
-	// Check if we should skip bootstrap (set by init container)
-	if _, err := os.Stat("/shared/skip-bootstrap"); err == nil {
+	skip := k8s.ShouldSkipBootstrap() // Kubernetes or bare-metal check.
```

Comment:
maybe use something other than `k8s.` as a namespace since the check applies both for k8s and bare-metal.